### PR TITLE
agent-ui: migrate to react router v4 and material ui 1 beta

### PR DESCRIPTION
**Pull request by t-bast**

### CODE REVIEW

File: packages/agent-ui/src/actions/findSegments.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { getAgent } from 'stratumn-agent-client';
+import { findSegmentsSuccess, findSegmentsFailure } from './';
+
+export default function(url, processName, options) {
+  return dispatch =>
+    getAgent(url)
+      .then(agent => {
+        const proc = agent.processes[processName];
```
Comment (jeremie-stratumn) [inline packages/agent-ui/src/actions/findSegments.js]: we can now use alexandre's getProcess function: https://github.com/stratumn/indigo-js/commit/704914b032034bdf62d7fd517e8c1a249e62822e

File: packages/agent-ui/src/actions/getMapIds.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { getAgent } from 'stratumn-agent-client';
+import { getMapIdsSuccess, getMapIdsFailure } from './';
+
+export default function(url, processName) {
+  return dispatch =>
+    getAgent(url)
+      .then(agent => {
+        const proc = agent.processes[processName];
```
Comment (jeremie-stratumn) [inline packages/agent-ui/src/actions/getMapIds.js]: same here, let's use getProcess https://github.com/stratumn/indigo-js/commit/704914b032034bdf62d7fd517e8c1a249e62822e

File: packages/agent-ui/src/components/processInfoPage.js [hunk 3]
```diff
@@ -19,10 +19,11 @@ export const ProcessInfoPage = ({ process }) => (
 );
 
 export function mapStateToProps(state, ownProps) {
+  const processName = ownProps.match.params.process;
```
Comment (jeremie-stratumn) [inline packages/agent-ui/src/components/processInfoPage.js]: You are missing a `withRouter` at the bottom of the page here:
`export default withRouter(connect(mapStateToProps)(ProcessInfoPage));`

Comment (t-bast) [reply packages/agent-ui/src/components/processInfoPage.js]: Nope I don't need it here. You only need withRouter if you're not rendered inside a route, and the process info page is always rendered inside a route so it has the routing info without needing withRouter

File: packages/agent-ui/src/actions/findSegments.js (hunk 1 above)
Comment (t-bast) [reply packages/agent-ui/src/actions/findSegments.js]: Good idea, on it

**Pull request merged**